src/mystery_gift.c

```diff
@@ -0,0 +1,293 @@
+#include "global.h"
+#include "mystery_gift.h"
+#include "save.h"
+#include "save_arrays.h"
+#include "constants/save_arrays.h"
+
+BOOL MysteryGiftTagIsValid(u32 tag);
+MysteryGift* SaveMysteryGift_GetByIdx(MysteryGiftSave* mg, int index);
+BOOL SaveMysteryGift_SetReceivedByIdx(MysteryGiftSave* mg, int index);
```

Comment:
pointer alignment on all these

Reply:
Not sure what you mean here

Reply:
pointer should be `MysteryGiftSave *mg` instead of current

Reply:
that's agreed by several people, and is the style in the whole repo

Reply:
it's ok red we can just fix it if pika wants to be dense here